File: typescript/src/01-strategy/after/phonepe-payment-strategy.ts
import type { IPaymentStrategy } from './payment-strategy.js';
import type { PaymentResult } from '../../shared/models.js';

// New provider: add a class without changing existing checkout code.
export class PhonePePaymentStrategy implements IPaymentStrategy {
  pay(amount: number): PaymentResult {
    console.log(`  [After] Charging ₹${amount} via PhonePe strategy`);
    return { success: true, message: 'PhonePe payment processed' };
  }
}

File: csharp/DesiArchitect.DesignPatterns/01_Strategy/After/PhonePePaymentStrategy.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Strategy.After;

public class PhonePePaymentStrategy : IPaymentStrategy
{
    public PaymentResult Pay(decimal amount)
    {
        Console.WriteLine($"  [After] PhonePe payment of ₹{amount} processed via PhonePe gateway");
        return new PaymentResult(true, "PhonePe payment successful");
    }
}
